Python Examples/TestMonitor/test_data_manager_client.py

```diff
@@ -0,0 +1,140 @@
+    :return: A dictionary which is required for updating the results
+    """
+    return{
+        "results": results,
+        "determineStatusFromSteps": determine_status_from_steps
+    }
+
+def create_results(results):
+    """
+    Creates new test results from the supplied models. The server automatically generates the result ids.
+    :param results: Results which needs to be created
+    :return: json response after creating the results
+    """
+    if len(results) == 0 :
+        raise ValueError("Number of results needs to be created can not be empty")
+    body = create_test_result_request(results)
+    request_uri = base_uri + create_results_route
+    request_response = raise_post_request(request_uri, body)
+
+    return request_response.json()
+
+def update_results(results):
+    """
+    Updates existing test results by merging or replacing values.
+    :param results: Results which needs to be updated
+    :return: json response after updating the results
+    """
+    if len(results) == 0 :
+        raise ValueError("Number of results needs to be updated can not be empty")
+    body = update_test_results_request(results, determine_status_from_steps=True)
+    request_uri = base_uri + update_results_route
+    request_response = raise_post_request(request_uri, body)
+
+    return request_response.json()
+
+def create_steps(steps):
+    """
+    Creates new test steps from the supplied models. The result associated with the step must exist prior to step creation. The server automatically generates step ids if not supplied.
+    :param steps: Steps which needs to be created
+    :return: json response after creating steps
+    """
+    if len(steps) == 0 :
+        raise ValueError("Number of steps needs to be created can not be empty")
+    body = test_step_create_or_update_request_object(
+            steps, update_result_total_time=True
+        )
+    request_uri = base_uri + create_steps_route
+    request_response = raise_post_request(request_uri, body)
+
+    return request_response.json()
+
+def update_steps(steps):
+    """
+    Updates existing steps by merging or replacing values.
+    :param steps: Steps which needs to be updated
+    :return: json response after updating the steps
+    """
+    if len(steps) == 0 :
+        raise ValueError("Number of steps needs to be updated can not be empty")
+    body = test_step_create_or_update_request_object(
+                steps, update_result_total_time=True
+            )
+    request_uri = base_uri + update_steps_route
+    request_response = raise_post_request(request_uri, body)
+
+    return request_response.json()
+
+def raise_post_request(uri, body):
+    """
+    Makes the post request API call.
+    :param uri: request uri which needs to be called
+    :param body: API call body
+    :return: json response of the api call
+    """
+    request_response =  requests.post(uri, json=body, headers=headers)
+    request_response.raise_for_status()
```

Comment:
The 200 OK response should be handled special. It indicates a partial success, but depending on the failure mode we can report partial success even when all the results in the request failed. 200 OK is also what update returns in a full success, whereas create will return 201. So you should instead inspect 200 OK responses for errors and raise in that case.

Reply:
Handled the partial success response by checking whether there is an error field in the response which indicates partial success response of the request in the main example file.